cm_service_manager.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+import smtplib
+import cm_api
+from cm_api.api_client import ApiResource
+
+parser = argparse.ArgumentParser(description='Restart a CM cluster service')
+
+parser.add_argument('-m', '--cm_host', required=True,
+                    help="hostname / fqdn of the CM UI ('cloudera.domain.com')")
+parser.add_argument('-u', '--username', default=os.environ['USER'],
+                    help="username for the CM API authentication ('johndoe')")
+parser.add_argument('-p', '--password', required=True,
+                    help="password for the CM API authentication ('Pa55w0rd')")
+parser.add_argument('-l', '--cluster', required=False,
+                    help="cluster to restart service in ('mycluster')")
+parser.add_argument('-s', '--service', required=False,
+                    help="service to restart ('hdfs')")
+parser.add_argument('-e', '--smtp_server', required=False,
+                    help="smtp server to send e-mail notifications")
+
+args = parser.parse_args()
+print args
+
+# https://cloudera.github.io/cm_api/docs/quick-start/
+api = ApiResource(args.cm_host, username=args.username, password=args.password)
+
+def select_from_items(item_type, items):
+    print("\n".join(["\nChoose a %s" % item_type] + [i.lower() for i in items]))
+    item = raw_input().lower()
+    print "\nInput received: %s " % item
+    if item in items:
+        return item
+    else:
+        print 'Bad input received...\n\n'
+        return select_from_items(item_type, items)
+
+
+# Rolling restart method (using few defaults - Role types, role names. It will take all)
+def rolling_restart(service_rr):
+    print service_rr
+    try:
+        #if service_rr.name.lower() in ['hive', 'hue', 'oozie', 'spark']:
+        #    raise AttributeError("Rolling Restart is not available for '%s' " % service.name.lower())
+        batch_size = raw_input("Number of worker roles to restart together in a batch \n")
+        fail_count_threshold = raw_input("Stop rolling restart if more than this number of worker batches fail \n")
+
+        options = ["true", "false"]    # options for stale config
+        stale_config_only = select_from_items('option to Restart roles on nodes with stale config?(Default is false) \n', options)   # Needs to be a user selection
+
+        rr_command = service_rr.rolling_restart(batch_size, fail_count_threshold, None, stale_config_only, None, None, None).wait()
+    except:
+        print "WARNING: Rolling Restart is not available for '%s'. Exiting... " % service.name.lower()
+        print "Executing Restart on service '%s' " % service.name.lower()
```

Comment:
But wasn't that the plan? Where the services which do not have RR functionality, will be just restarted over here. 
I am coming to this , since the _user_ is selecting that particular service and aims to cycle it. 
So instead, can it be like follows:

User  selects a service which doesn't support RR, so it will error out in the try block. 
When it goes to the except block, that time we give a prompt "Instead do a Normal restart? Yes or No"
The user has the option here only to continue or not. Instead of starting from the beginning.